File: specs/NativeTTSModule.ts
import type {TurboModule} from 'react-native';
import {TurboModuleRegistry} from 'react-native';

export interface Spec extends TurboModule { 
  speak(language: string, text: string): void; // 텍스트를 음성으로 출력
}

export default TurboModuleRegistry.getEnforcing<Spec>(
  'NativeTTSModule',
);

File: android/app/src/main/java/com/nativettsmodule/NativeTTSModule.kt
package com.nativettsmodule

import android.speech.tts.TextToSpeech
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactMethod
import com.nativettsmodule.NativeTTSModuleSpec
import java.util.Locale

class NativeTTSModule(reactContext: ReactApplicationContext) : NativeTTSModuleSpec(reactContext) {

  private var tts: TextToSpeech? = null

  init {
    tts = TextToSpeech(reactContext) { status -> }
  }

  override fun getName() = NAME

  @ReactMethod
  override fun speak(language: String, text: String) {
    val locale = Locale(language)
    tts?.setLanguage(locale)
    tts?.speak(text, TextToSpeech.QUEUE_FLUSH, null, "TTS_ID")
  }

  override fun invalidate() {
    tts?.shutdown()
  }

  companion object {
    const val NAME = "NativeTTSModule"
  }
}
